Source/DataAccessLayer/Bases/EntityProviderBaseCore.generated.cst

```diff
@@ -83,1310 +83,1369 @@ using <%=DALNameSpace%>;
-		/// </summary><%=TransformStoredProcedureInputsToMethodComments(command.InputParameters) + TransformStoredProcedureOutputsToMethodComments(command.AllOutputParameters)%>
-		/// <param name="start">Row number at which to start reading, the first row is 0.</param>
-		/// <param name="pageLength">Number of rows to return.</param>
-		/// <param name="transactionManager"><see cref="TransactionManager"/> object</param>
-		/// <remark>This method is generate from a stored procedure.</remark><% if (returnType != "void") {%>
-		/// <returns>A <see cref="<%=returnTypeForComment%>"/> instance.</returns><%}%>
-		public abstract <%=returnType%> <%=methodName%>(TransactionManager transactionManager, int start, int pageLength <%=TransformStoredProcedureInputsToMethod(true, command.InputParameters) + TransformStoredProcedureOutputsToMethod(true, command.AllOutputParameters)%>);
-		
-		#endregion
-	<%
-		//}
-		} // end foreach command	
-	%>
-		
-		#endregion
+    %>
+        
+        #region <%=command.Name%> 
+        
+        /// <summary>
+        ///    This method wrap the '<%=command.Name%>' stored procedure. 
+        /// </summary><%=TransformStoredProcedureInputsToMethodComments(command.InputParameters) + TransformStoredProcedureOutputsToMethodComments(command.AllOutputParameters)%>
+        /// <remark>This method is generate from a stored procedure.</remark><% if (returnType != "void") {%>
+        /// <returns>A <see cref="<%=returnTypeForComment%>"/> instance.</returns><%}%>
+        public <%=returnType%> <%=methodName%>(<%=TransformStoredProcedureInputsToMethod(false, command.InputParameters) + TransformStoredProcedureOutputsToMethod(command.InputParameters.Count > 0, command.AllOutputParameters)%>)
+        {
+            <%= returnKeyword %> <%=methodName%>(null, 0, int.MaxValue <%=TransformStoredProcedureInputsToDataAccess(true, command.InputParameters) + TransformStoredProcedureOutputsToDataAccess(true, command.AllOutputParameters)%>);
+        }
+        
+        /// <summary>
+        ///    This method wrap the '<%=command.Name%>' stored procedure. 
+        /// </summary><%=TransformStoredProcedureInputsToMethodComments(command.InputParameters) + TransformStoredProcedureOutputsToMethodComments(command.AllOutputParameters)%>
+        /// <param name="start">Row number at which to start reading, the first row is 0.</param>
+        /// <param name="pageLength">Number of rows to return.</param>
+        /// <remark>This method is generate from a stored procedure.</remark><% if (returnType != "void") {%>
+        /// <returns>A <see cref="<%=returnTypeForComment%>"/> instance.</returns><%}%>
+        public <%=returnType%> <%=methodName%>(int start, int pageLength<%=TransformStoredProcedureInputsToMethod(true, command.InputParameters) + TransformStoredProcedureOutputsToMethod(true, command.AllOutputParameters)%>)
+        {
+            <%= returnKeyword %> <%=methodName%>(null, start, pageLength <%=TransformStoredProcedur
```

Comment:
Sorry, forgot one comment.  you added a new parameter right?  Shouldn't there be a default valid for this so existing code does not break?  I would think you would default that to -1 or make it a nullable int and make it null.  That way existing code that is using it doesn't have compile errors out of the box.

Reply:
I added two new overloads.  The oldest supported version of .net does not have default values, so I could not use those.
I also moved one of the deep load methods down to be in order with the rest of them, for some reason it was the only one out of order.